### PR TITLE
Only validate honor mode for non-prof-ed courses.

### PR DESCRIPTION
**Pull request by peter-fogg**
Quick fix for [ECOM-3170](https://openedx.atlassian.net/browse/ECOM-3170).

@bderusha @rlucioni

### CODE REVIEW

File: ecommerce/static/js/models/course_model.js [hunk 1]
```diff
@@ -62,7 +62,9 @@ define([
                     msg: gettext('You must select a course type.')
                 },
                 honor_mode: {
-                    required: true,
+                    required: function () {
+                        return this.get('type') !== 'professional';
```
Comment (bderusha) [inline ecommerce/static/js/models/course_model.js]: Can we use the `includeHonorMode` method here to reduce duplicate code?

**Comment by rlucioni** [comment]
:+1:

File: ecommerce/static/js/models/course_model.js (hunk 1 above)
Comment (peter-fogg) [inline ecommerce/static/js/models/course_model.js]: Good idea, that's done.

**Comment by bderusha** [comment]
:+1:

**Pull request merged**